### DISCUSSION TITLE
API for quering the currently applied filters on the dashboard

### DISCUSSION THREAD

**Discussion by reedasha** [apache/superset#25609] [Q&A / Help]
Hi,

So I'm embedding Superset through iframe into the React application and what I want to do is to use filters, either the filters on the left hand side or drill down into charts and then once I filtered the data needed, I want to retrieve all the filters I applied on this dashboard through an API call. I need the currently applied filters to send it over to our own backend server and perform operation on that particular filtered dataset.

Is there already a possibility to do it? 

I know there's this API 
![image](https://github.com/apache/superset/assets/16371152/1ec167e6-54cb-4d8b-b18f-1827f5a28992)
but it only seems to returning the filters applied from the left hand side filter menu, not the drilldown filters I applied through the charts. Plus I don't know whats the way to retrieve the filter key, when superset is embedded into an app.

Any suggestions are much appreciated.
Thanks

**Comment by vnayagam** [comment]
I too have the same question, Is there a way to get filter state key ?

**Reply by darrylhebbes** [reply]
The response of the POST and PUTs contain the key. Check the swagger docs response schema

**Comment by erdogankesgin** [comment]
Has anyone found any solution?

**Comment by vikashrajgupta** [comment]
@rusackas @andy-clapson - Can you please help me with this? I think you're the most important person here who can help to provide a solution if it's possible in Superset.

I have also similar requirements, basically having to fetch all of the filters that the user will apply inside the Iframe dashboard that are fetched using embedded SDK.

I tried one of them `http://localhost:8080/api/v1/dashboard/45/filter_state/1RrizUUqS-itYHcKutFbO-U3P9bELVgiNAcyhhnkK6U6f5oHfA-hhaQnjObnni7s` it's working fine but from the interface only.

That would be great if I could get something similar when I'm using a dashboard in Iframe.

**Comment by rameshsinghds** [comment]
I also have a requirement to get the Cross Filters applied in a dashboard. Is there any simple solution to that?

**Reply by darrylhebbes** [reply]
With an auth_token (check superset user role) you can make a POST to `/api/v1/dashboard/{id}/filter_state` with the filter changes in the POST body. See how Superset itself does this by inspecting the traffic when setting filters.
This will return you the filter key in the response if 200, with which you can use to embed using the SDK.
The filter key is recreated on each POST/PUT call

Let me warn you, nothing about Superset cross filters is simple.
